make push fails because gcr.io is hardcoded into push script

<!--
Please see https://istio.io/troubleshooting for potential workarounds.
Please add the correct labels and epics (and priority and milestones if you have that information)
-->
```The push refers to a repository [docker.io/sdake/sidecar_initializer]
315c1b239616: Pushed 
sdake: digest: sha256:57d1aaaf876c31134b7a89ef80591d51515870a9a3a8f25ee01cb3a0cc562ac1 size: 528
+ for IMAGE in '${IMAGES[@]}'
+ for TAG in '${TAGS[@]}'
+ for HUB in '${HUBS[@]}'
+ docker tag eurekamirror docker.io/sdake/eurekamirror:sdake
+ '[' false '!=' true ']'
+ docker_push docker.io/sdake/eurekamirror:sdake
+ local im=docker.io/sdake/eurekamirror:sdake
+ [[ docker.io/sdake/eurekamirror:sdake =~ ^gcr\.io ]]
+ docker push docker.io/sdake/eurekamirror:sdake
The push refers to a repository [docker.io/sdake/eurekamirror]
8071006dac11: Pushed 
sdake: digest: sha256:f47f523fe61448ba33df677dd86fd38e73dfc3d98385b429bad14486b0e4275f size: 528
+ for job in '`jobs -p`'
+ echo 18058
18058
+ wait 18058
+ for job in '`jobs -p`'
+ echo 18059
18059
+ wait 18059
+ for job in '`jobs -p`'
+ echo 18060
18060
+ wait 18060
+ let FAIL+=1
+ for job in '`jobs -p`'
+ echo 18061
18061
+ wait 18061
+ let FAIL+=1
+ for job in '`jobs -p`'
+ echo 18062
18062
+ wait 18062
+ let FAIL+=1
+ '[' 3 '!=' 0 ']'
+ echo 'FAIL! (3)'
FAIL! (3)
+ exit 3
Makefile:62: recipe for target 'push' failed
make: *** [push] Error 3
```

To reproduce:
```make setup
make build
sudo -E make docker
sudo -E make push
```

--- a/bin/push
+++ b/bin/push
@@ -10,9 +10,13 @@ cd $ROOT
 pilot/bin/push-docker -hub ${HUB} -tag ${TAG} &
 mixer/bin/push-docker -hub ${HUB} -tag ${TAG} &
 security/bin/push-docker -hub ${HUB} -tag ${TAG} &
-pilot/bin/upload-istioctl -p "gs://istio-artifacts/pilot/${TAG}/artifacts/istioctl" &
-pilot/bin/push-debian.sh -c opt -p "gs://istio-artifacts/pilot/${TAG}/artifacts/debs" &
-security/bin/push-debian.sh -c opt -p "gs://istio-artifacts/auth/${TAG}/artifacts/debs" &
+
+# Permit $HUB overrides to suppress errors during push
+if [[ "${HUB}" =~ ^gcr\.io ]]; then
+    pilot/bin/upload-istioctl -p "gs://istio-artifacts/pilot/${TAG}/artifacts/istioctl" &
+    pilot/bin/push-debian.sh -c opt -p "gs://istio-artifacts/pilot/${TAG}/artifacts/debs" &
+    security/bin/push-debian.sh -c opt -p "gs://istio-artifacts/auth/${TAG}/artifacts/debs" &
+fi
 
 FAIL=0
 for job in `jobs -p`

